File: cpp/DoublePoint/maxArea/maxArea.cpp
#include <vector>

class Solution {
  public:
    int maxArea(std::vector<int> &height) {
        int left = 0;
        int right = height.size() - 1;
        int maxArea = 0;
        while (left < right) {
            int area = std::min(height[right], height[left]) * (right - left);
            maxArea = std::max(maxArea, area);
            if (height[left] <= height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return maxArea;
    }
};

File: golang/DoublePoint/maxArea/maxArea.go
// Package maxarea leetcode 11.盛最多水的容器
package maxarea

func maxArea(height []int) int {
	left := 0
	right := len(height) - 1
	maxArea := 0

	for left < right {

		area := min(height[left], height[right]) * (right - left)

		maxArea = max(maxArea, area)

		if height[right] > height[left] {
			left++
		} else {
			right--
		}
	}

	return maxArea
}
